libs/librepcb/common/fileio/transactionalfilesystem.cpp

```diff
@@ -68,8 +68,8 @@ TransactionalFileSystem::TransactionalFileSystem(const FilePath& filepath,
     if (restoreMode == RestoreMode::ASK) {
       QMessageBox::StandardButton btn = QMessageBox::question(
           0, tr("Restore autosave backup?"),
-          tr("It seems that the application was crashed the last time. "
-             "Do you want to restore the last autosave backup?"),
+          tr("It seems that the application crashed the last time you opened "
+             "this project. Do you want to restore the last autosave backup?"),
```

Comment:
Oh, I just realized that this change is actually not fully valid :( The class `TransactionalFileSystem` is not only used for projects, but also for libraries. So the message should be kept more generic. I'm so sorry for missing that point in my last review, that wasn't my intention :weary:

Reply:
I wonder if we could parametrize the message somehow, so that it's more clear?

Reply:
Hmm actually I don't like to open a `QMessageBox` in such a low level class anyway, thus I quickly tried to implement a different concept: Instead of passing an enum for choosing the restore mode (yes/no/ask/abort), a callback can be passed instead. The callback then returns a bool to choose whether the backup should be restored or not. Then the message box has to be implemented in a higher level class, and thus could be more specific.

My suggestion: We merge this PR as-is and afterwards I'll open a PR to apply the refactoring - do you agree @dbrgn?

Reply:
Yep sounds great! Opening message boxes from low-level classes is a bad idea indeed :)